Portal 1.16.0 RNs.mdx (#1053)
TykTechnologies/tyk-docs: developer-support/release-notes/portal.mdx

---
title: "Tyk Enterprise Developer Portal Release Notes"

> description: "Release notes documenting updates, enhancements and changes for Tyk Enterprise Developer Portal v1.13.0"

Reviewer: ```suggestion
description: "Release notes documenting updates, enhancements and changes for Tyk Enterprise Developer Portal"
```

---

## 1.16 Release Notes

### 1.16.0 Release Notes

#### Release Date xx December 2025

#### Release Highlights

> In this release, we’ve introduced a new Direct Access Flow as an alternative to the shopping cart flow, and we've enabled single credential management for multiple API products, along with enhanced SSO integration, improved mTLS certificate management, and several improvements and security fixes. 

Reviewer: ```suggestion
Developer Portal v1.16.0 addresses credential sprawl that impacts large-scale API programs. Developers can now reuse a single credential across multiple API products. We've also streamlined the access request flow from 5 steps to 2. For admins, you get instant central control: add or remove API access from any credential in seconds, change plans without generating new keys, and manage mTLS certificates centrally for compliance.

**Key benefits:**
- **One key, multiple APIs:** Reuse credentials when auth requirements match
- **Streamlined access requests:** Direct flow eliminates the shopping cart (now optional)
- ** Change plans on existing credentials:** Upgrade rate limits without regenerating keys or redeploying apps
- ** Central credential management:** Add/remove products or revoke access via Portal admin, without need to contact developers
- **Managed mTLS certificates:** Centralize certificate control for compliance while developers self-serve access requests```

#### Upgrade instructions

If you are on version 1.15.0 or an older version, we recommend upgrading to this release ASAP.

To upgrade the Portal's theme, please follow the [upgrade instructions](/portal/customization/themes#upgrading-themes).

#### Download
- [Docker image v1.16.0](https://hub.docker.com/r/tykio/portal/tags?page=&page_size=&ordering=&name=v1.16.0)
  - ```bash
    docker pull tykio/portal:v1.16.0
    ```

- [The default theme package](https://github.com/TykTechnologies/portal-default-theme/releases/tag/1.16.0)

#### Changelog
<a id="Changelog-v1.16.0"></a>

##### Changed
<AccordionGroup>

<Accordion title='Updated External Portal UI Design and Branding'>
The External Developer Portal now features a refreshed visual design with updated brand colors, modern typography, and improved spacing. Users will experience a cleaner, more professional interface with enhanced readability and consistent design elements across all portal pages, while maintaining full functionality and accessibility standards. Furthermore, the API catalogue has been redesigned with new card layouts for users to experience a cleaner and more professional interface.
</Accordion>

> </AccordionGroup>

Reviewer: Move it after "Added" section

Reviewer 2: IMO, the `changed` section is the correct place as we haven't added anything new.

Reviewer: I mean normally we have "Added" section before "Changed" section in a change log

##### Added

<AccordionGroup>

> <Accordion title='Direct Access Flow for API Product Requests'>

Reviewer: ```suggestion
<Accordion title='Direct Access Flow for API Products'>
```

> The Developer Portal now offers a streamlined Direct Access Flow that can substitute (if enabled in Admin General Settings) the shopping cart flow for API product access requests. This new flow allows developers to request access to API products immediately without adding them to a cart, reducing friction and improving the developer experience.

Reviewer: ```suggestion
The Developer Portal now offers a streamlined Direct Access Flow that can substitute (if enabled in Admin General Settings) the Cart-based flow for API product access requests. This new flow allows developers to request access to API products immediately without adding them to a cart, reducing friction and improving the developer experience.
```

> For more information, please visit [Requesting Access to an API Product](https://tyk.io/docs/product)

Reviewer: Link TBD

</Accordion>

<Accordion title='Single Credential for Multiple API Products'>

> Developers can now add multiple API products to a single credential set, eliminating credential sprawl. This feature allows for better credential management while maintaining security and access control.

Reviewer: ```suggestion
Developers can now add multiple API products to a single credential, eliminating credential sprawl and simplifying key management. This feature allows for simpler credential management while maintaining security and access control.
```

> For more information, please visit [Credential Management and Reusability](https://tyk.io/docs/product)

Reviewer: Link TBD

</Accordion>

> 

Reviewer: ```suggestion

<Accordion title='Plan Management for Existing Credentials'>
Both API consumers and portal admins can now change plans for products on active credentials within an app, allowing access level updates without regenerating keys or redeploying apps.
</Accordion>

```

<Accordion title='Custom Login URL Support'>
The Developer Portal now supports configuring a custom login URL that redirects users to external identity providers instead of the default password login page. When configured, all authentication flows automatically redirect to the specified external login system. This can be configured by adding the `SSOCustomLoginURL` setting to your portal configuration or setting the environment variable `PORTAL_SSO_CUSTOM_LOGIN_URL` when starting the portal.

> </Accordion>

Reviewer: ```suggestion
<Accordion title='Custom SSO Login URL Configuration'>
The Developer Portal now supports configuring a custom SSO login URL that redirects users to your identity provider's login page instead of the default Portal login page. When configured, all authentication flows automatically redirect to the specified external login system. This can be configured by adding the `SSOCustomLoginURL` setting to your portal configuration or setting the environment variable `PORTAL_SSO_CUSTOM_LOGIN_URL` when starting the portal.
</Accordion>
```

Reviewer: Make sure portal configurations page is also updated: https://tyk-joancamostyk-patch-13.mintlify.app/product-stack/tyk-enterprise-developer-portal/deploy/configuration

> <Accordion title='Centralised Management for mTLS APIs'>

Reviewer: ```suggestion
<Accordion title='Centralized Management for mTLS APIs'>
```

> API owners can now configure how client certificates are managed for mTLS-secured APIs in their API Products. Instead of always requiring developers to upload certificates during access requests, API owners can choose between self-serve certificate management (current behavior) or centralised managed certificates, where certificates are pre-configured via the Dashboard, and API Consumers will not be requested (nor able) to provide a certificate when requesting access.

Reviewer: @sharadregoti potential a new guide for configuring "managed" mtls cert for access

</Accordion>

<Accordion title='Added Developer App Credentials Access on Product Documentation Pages'>
Enhanced the Developer Portal by making developer app credentials available directly within API product documentation pages. This eliminates the need for users to navigate away from documentation to retrieve their API keys, significantly reducing friction when testing APIs.

> </Accordion>

Reviewer: ```suggestion
<Accordion title='Exposed Developer App Credentials Access on Product Pages'>
Exposed developer app credentials on product detail and documentation pages, making it possible to inject approved credentials to OpenAPI and GraphQL playground. Integrating the credentials to your templates could eliminate the need for users to navigate away from documentation to retrieve their API keys, significantly reducing friction when testing APIs.
</Accordion>
```

@tbuchaillot @flor need review on this

Reviewer 3: LGTM

<Accordion title='Enhanced GraphQL URL Handling for Trailing Slash Consistency'>
Improved GraphQL endpoint URL handling to automatically support both trailing slash and non-trailing slash variations, ensuring seamless compatibility when users copy API endpoints from the Dashboard to the GraphQL server URL in the Developer Portal.
</Accordion>

> 

Reviewer: ```suggestion

```
Removing this, no change was done to handle trailing slash in https://tyktech.atlassian.net/browse/TT-16184

> <Accordion title='Added Auto-Population of GraphQL Server Endpoints in API Product Documentation'>

Reviewer: ```suggestion
<Accordion title='Added Auto-Population of GraphQL Server URL in API Product Documentation'>
```

> Enhanced the API Product creation workflow by automatically pre-populating the GraphQL server endpoint field when a GraphQL API is selected. The system now combines the Gateway host with the API's Listen Path to generate the correct endpoint format, while keeping the field editable for custom configurations.

Reviewer: ```suggestion
Enhanced the API Product creation workflow by automatically pre-populating the GraphQL server URL field when a GraphQL API is selected. The system now combines the baseline URL of the provider with the API's listen path to generate the correct URL, while keeping the field editable for custom configurations.

If the BaselineURL of the provider is not configured, `<< gateway url>>` will be used as placeholder.
```

#### Changelog
<a id="Changelog-v1.16.0"></a>

##### Changed
<AccordionGroup>

<Accordion title='Updated External Portal UI Design and Branding'>
The External Developer Portal now features a refreshed visual design with updated brand colors, modern typography, and improved spacing. Users will experience a cleaner, more professional interface with enhanced readability and consistent design elements across all portal pages, while maintaining full functionality and accessibility standards. Furthermore, the API catalogue has been redesigned with new card layouts for users to experience a cleaner and more professional interface.
</Accordion>

</AccordionGroup>

##### Added

<AccordionGroup>

<Accordion title='Direct Access Flow for API Products'>
The Developer Portal now offers a streamlined Direct Access Flow that can substitute (if enabled in Admin General Settings) the Cart-based flow for API product access requests. This new flow allows developers to request access to API products immediately without adding them to a cart, reducing friction and improving the developer experience.

> For more information, please visit [Requesting Access to an API Product](https://tyk.io/docs/product)

Reviewer 2: ```suggestion
For more information, please visit [Requesting Access to an API Product](/tyk-developer-portal/direct-access-flow)
```

</Accordion>

<Accordion title='Single Credential for Multiple API Products'>
Developers can now add multiple API products to a single credential, eliminating credential sprawl and simplifying key management. This feature allows for simpler credential management while maintaining security and access control.

> For more information, please visit [Credential Management and Reusability](https://tyk.io/docs/product)

Reviewer 2: ```suggestion
For more information, please visit [Credential Management and Reusability](/tyk-developer-portal/single-credentials-multiple-api-products)
```

---

## 1.16 Release Notes

### 1.16.0 Release Notes

#### Release Date xx December 2025

#### Release Highlights

Developer Portal v1.16.0 addresses credential sprawl that impacts large-scale API programs. Developers can now reuse a single credential across multiple API products. We've also streamlined the access request flow from 5 steps to 2. For admins, you get instant central control: add or remove API access from any credential in seconds, change plans without generating new keys, and manage mTLS certificates centrally for compliance.

**Key benefits:**
- **One key, multiple APIs:** Reuse credentials when auth requirements match
- **Streamlined access requests:** Direct flow eliminates the shopping cart (now optional)
- ** Change plans on existing credentials:** Upgrade rate limits without regenerating keys or redeploying apps
- ** Central credential management:** Add/remove products or revoke access via Portal admin, without need to contact developers

> - **Managed mTLS certificates:** Centralize certificate control for compliance while developers self-serve access requests```

Reviewer: ```suggestion
- **One key, multiple APIs:** Reuse credentials when auth requirements match
- **Streamlined access requests:** Direct flow eliminates the shopping cart (now optional)
- **Change plans on existing credentials:** Upgrade rate limits without regenerating keys or redeploying apps
- **Central credential management:** Add/remove products or revoke access via Portal admin, without need to contact developers
- **Managed mTLS certificates:** Centralize certificate control for compliance while developers self-serve access requests
```

</Accordion>

<Accordion title='GraphQL Documentation and Playground Support for UDG APIs'>
Fixed Developer Portal to properly handle Universal Data Graph (UDG) APIs as GraphQL APIs. Portal admins can now upload SDL (Schema Definition Language) files and configure GraphQL server URLs for UDG APIs instead of being forced to use OpenAPI Specification (OAS) documentation. API consumers can now access the GraphQL playground when enabled, allowing proper interaction with UDG APIs through the native GraphQL interface.
</Accordion>

<Accordion title='Fixed Policy Override Issue When Updating Products in Developer Portal'>
Resolved issue where the Developer Portal was overriding or reverting policy changes made through the Tyk Dashboard when updating API Products. The Portal now preserves existing policy configurations, including Path-Based Permissions and ACL settings, instead of overwriting the entire policy object.
</Accordion>

</AccordionGroup>

##### Security Fixes

<AccordionGroup>

> <Accordion title='Private Catalogue Documentation Access Control'>

Reviewer 2: @alienor, is this ok to expose?

Reviewer 4: @sharadregoti - I would remove.

Reviewer 2: @JoanCamosTyk, please remove it. Also, refer to this [document](https://tyktech.atlassian.net/wiki/spaces/securitycop/pages/2807857178/Public+Release+Notes) on what line items to be included in the security section

---

## 1.16 Release Notes

### 1.16.0 Release Notes

#### Release Date xx December 2025

#### Release Highlights

> Developer Portal v1.16.0 addresses credential sprawl that impacts large-scale API programs. Developers can now reuse a single credential across multiple API products. We've also streamlined the access request flow from 5 steps to 2. For admins, you get instant central control: add or remove API access from any credential in seconds, change plans without generating new keys, and manage mTLS certificates centrally for compliance.

Reviewer: ```suggestion
Developer Portal v1.16.0 addresses credential sprawl that impacts large-scale API programs. Developers can now reuse a single credential across multiple API products. We've also streamlined the access request flow from 5 steps to 2. For admins, you get instant central control: add or remove access to API product from any credential in seconds, change plans without generating new keys, and manage mTLS certificates centrally for compliance.
```

**Key benefits:**
- **One key, multiple APIs:** Reuse credentials when auth requirements match

> - **Streamlined access requests:** Direct flow eliminates the shopping cart (now optional)

Reviewer: ```suggestion
- **Streamlined access requests:** Direct access flow eliminates the need to add API products to shopping cart
```

- **Change plans on existing OAuth 2.0 credentials:** Upgrade rate limits without regenerating keys or redeploying apps
- **Central credential management:** Add/remove products or revoke access via Portal admin, without need to contact developers
- **Managed mTLS certificates:** Centralize certificate control for compliance while developers self-serve access requests

For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.16.0) below.

#### Breaking Changes

There are no breaking changes in this release.

#### Known Issues

##### Quota Counter Reset When Adding API Products to Existing Credentials

> When adding an API Product to an existing Auth (Bearer) Token credential for the first time, the quota usage counters for that credential may be reset to zero.

Reviewer: ```suggestion
When adding an API Product to an existing Auth (Bearer) Token credential for the first time, the quota usage counters for that credential may be reset to zero, allowing temporary quota bypass until next reset cycle. 
```

**Details:**
- Affects Auth (Bearer) Token authentication only
- Occurs only on the first API product addition
- Subsequent API product additions are not affected

> - Quota counters automatically return to normal behavior after the next quota reset cycle (hourly/daily)

Reviewer: @asutosh @alienor should we disclose this?

Reviewer 5: I wouldn't give the detail here - just acknowledge "some limitations"... and then put the detail in the internal sales/support enablement docs.
